src/dataflow/v2/controlflow-version-two.ts

```diff
@@ -0,0 +1,87 @@
+import { Expression } from 'xpath-ts2'
+import { Environment } from '../v1'
+
+export enum ExecutionState{
+	Always,
+	Maybe,
+	Never
+}
+
+
+export interface IExecutionTuple {
+	readonly executed:               ExecutionState,
+	readonly influencingExpressions: Expression[] //TODO: correct?
+}
+
+export class ExecutionTuple implements IExecutionTuple{
+	readonly executed:               ExecutionState
+	readonly influencingExpressions: Expression[]
+	constructor(executed:ExecutionState, influencingExpressions: Expression[]){
+		this.executed = executed
+		this.influencingExpressions = influencingExpressions
+	}
+}
+
+export function expressionExecution(previousExecutionState: ExecutionState, environmentUntilNow: Environment, currentExpression: Expression, _callStack: Expression[], _influencingExpressions: Expression[]){
+	//Set the State in which the Expression was found in
+	//Expression.entry = previousExecutionState
+	//TODO: how to save? efficient way unknown
+    
+	if(previousExecutionState == ExecutionState.Never){
+		//Expression.exit = new ExecutionTuple(previousExecutionState, influencingExpressions);
+		return
+	}
+
+	const currentExpressionType = evaluateExpressionType(currentExpression, environmentUntilNow)
+
+	switch(currentExpressionType){
+		case ExpressionType.Literal:
+		case ExpressionType.If:
+		case ExpressionType.Loop:
+		case ExpressionType.Function:
+		case ExpressionType.Then:
+		case ExpressionType.Else:
+			//TODO how to save
+			//Expr.exit = (ALWAYS, List(empty))
+			break
+		case ExpressionType.Break:
```

Comment:
Das brauchen wir so eigentlich nicht mehr. Wenn du magst kannst du dir natürlich ein mapping layer definieren was das abstrahiert, aber was du suchst ist in der `r-bridge` im `model` unter `type.ts` als `RType`. Da sind alle drin die es gibt. Da werd ich vermutlich bald noch ne view dazupacken die alle Typen enthält die wir jetzt dann in dataflow v2 überhaupt noch haben (also quasi: https://github.com/Code-Inspect/flowr/issues/568).